### PR TITLE
get_all_bins

### PR DESCRIPTION
**Pull request by graybri3**
This is the get_all_bins implementation that replaces the previous get_next_bin implementation.

### CODE REVIEW

**Comment by paulromano** [comment]
Thanks for the pull request, @graybri3. It looks like you inadvertently added some files to the repository. Can you remove them by running:
```bash
git rm .DS_Store
git rm tests/.DS_Store
git rm src/Makefile
git rm src/DartConfiguration.tcl
git rm src/CTestTestfile.cmake
git checkout develop examples/jupyter/pincell.ipynb
git commit
```

There's also a conflict with the develop branch because of the recent merging of #893 which added a new 'cell from' filter and modified the surface filter. You'll need to merge the develop branch into yours and resolve the conflicts. If you don't have a remote set up for the main OpenMC repo, you can set it up and merge by running:
```bash
git remote add upstream https://github.com/mit-crpg/openmc
git fetch upstream
git merge upstream/develop
```
At that point, you'll need to edit tally_filter.F90 to fix the conflict and update the 'cell from' filter. The get_all_bins subroutine for the 'cell from' filter should look very similar to what you have for the cell filter. Once you're done, you can commit and push.

**Comment by graybri3** [comment]
I updated the tally.F9 and the tally_filter.F90 files.  Hopefully it is fixed now.

File: src/tally_filter.F90 [hunk 1]
```diff
@@ -946,29 +841,23 @@ end function text_label_cellborn
...
-    next_bin = NO_BIN_FOUND
-    weight = ERROR_REAL
-    if (current_bin == NO_BIN_FOUND) then
       do i = 1, this % n_bins
-        if (abs(p % surface) == this % surfaces(i)) then
-          next_bin = i
-          weight = sign(1, p % surface)
+        if (p % surface == this % surfaces(i)) then
+          call match % bins % push_back(i)
+          call match % weights % push_back(ONE)
```
Comment (paulromano) [inline src/tally_filter.F90]: I'd suggest changing this here to:
```Fortran
if (abs(p % surface) == this % surfaces(i)) then
  call match % bins % push_back(i)
  if (p % surface < 0) then
    call match % weights % push_back(-ONE)
  else
    call match % weights % push_back(ONE)
  end if
  exit
end if
```
Pull request #893 changed the behavior for surface filters (hence why the `abs()` is now needed when comparing to the specified surfaces).

**Comment by paulromano** [comment]
Looks like you just have one test that is failing; my suggested change in `tally_filter.F90` should resolve that.

**Comment by graybri3** [comment]
I made those changes to tally_filter.F90.

**Comment by paulromano** [comment]
Thanks @graybri3!

**Pull request merged**